src/mad_spark_alt/evolution/cost_estimator.py

```
@@ -0,0 +1,658 @@
+        if model not in self._model_costs:
+            # Fall back to GPT-4 pricing if model not found
+            model = "gpt-4"
+
+        model_costs = self._model_costs[model]
+
+        if assume_equal_input_output:
+            # For simplicity, assume roughly equal input/output split
+            input_tokens = tokens // 2
+            output_tokens = tokens - input_tokens
+            return model_costs.calculate_cost(input_tokens, output_tokens)
+        else:
```

Ruff (0.12.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)